src/js/components/event/TodoList.js

```diff
@@ -0,0 +1,118 @@
+import { KEY } from "../../constants/eventKey.js";
+import { PRIORITY_VALUE, PRIORITY } from "../../constants/constant.js";
+
+async function onDeleteItem(event) {
+  const teamId = this.teamData._id;
+  const memberIndex = event.target.dataset.memberindex;
+  const memberId = this.memberListData[memberIndex].id;
+  const itemId = event.target.dataset.itemid;
+
+  const { error } = await fetchRequest(
+    API_URL.MEMBER_ITEM(teamId, memberId, itemId),
+    METHOD.DELETE
+  );
+  if (error) return alert(ERROR_MESSAGES.DELETE_ITEM);
+
+  this.memberListData[memberIndex].todoList = this.memberListData[memberIndex].todoList.filter(
+    (item) => {
+      if (item.id !== itemId) {
+        return item;
+      }
+    }
+  );
+
+  this.render(this.memberListData);
+}
+
+async function onCompleteItem(event) {
+  const teamId = this.teamData._id;
+  const memberIndex = event.target.dataset.memberindex;
+  const memberId = this.memberListData[memberIndex].id;
+  const itemId = event.target.dataset.itemid;
+
+  const { error } = await fetchRequest(API_URL.ITEM_TOGGLE(teamId, memberId, itemId), METHOD.PUT);
+
+  if (error) return alert(ERROR_MESSAGES.COMPLETE_ITEM);
+
+  this.memberListData[memberIndex].todoList.map((item) => {
+    item.id === itemId && (item.isCompleted = !item.isCompleted);
+  });
+
+  this.render(this.memberListData);
+}
+
+function onEditingItem(event) {
+  if (event.target.className !== "label") return;
+
+  const memberIndex = event.target.dataset.memberindex;
+  const itemId = event.target.dataset.itemid;
+
+  this.memberListData[memberIndex].todoList.map((item) => {
+    item.id === itemId && (item.editing = !item.editing);
+  });
+
+  this.render(this.memberListData);
+}
+
+async function onEditItem(event) {
+  const teamId = this.teamData._id;
+  const memberIndex = event.target.dataset.memberindex;
+  const memberId = this.memberListData[memberIndex].id;
+  const itemId = event.target.dataset.itemid;
+
+  if (event.key === KEY.ESCAPE) {
+    this.memberListData[memberIndex].todoList.map((item) => {
+      item.id === itemId && (item.editing = !item.editing);
+    });
+
+    this.render(this.memberListData);
+  }
+  if (event.key === KEY.ENTER) {
+    const { response, error } = await fetchRequest(
+      API_URL.MEMBER_ITEM(teamId, memberId, itemId),
+      METHOD.PUT,
+      { contents: event.target.value }
+    );
+
+    if ((response, error)) return alert(ERROR_MESSAGES.EDIT_ITEM);
+
+    this.memberListData[memberIndex].todoList.map((item) => {
+      if (item.id === itemId) {
+        item.contents = response.contents;
+        item.editing = !item.editing;
+      }
+    });
+
+    this.render(this.memberListData);
+  }
+}
+
+async function onSetPriority(event) {
+  if (event.target.value == 0) return;
+
+  const teamId = this.teamData._id;
+  const memberIndex = event.target.dataset.memberindex;
+  const memberId = this.memberListData[memberIndex].id;
+  const itemId = event.target.dataset.itemid;
```

Comment:
다른 함수의 코드에서도 id, index를 추출해야하는 경우가 많이 있는데 이 부분을 함수로 변경해보면 어떨까요?
```suggestion
  const { teamId, memberIndex, memberId, itemId } = getTargetInfo()???
```
처럼요!

Reply:
함수로 빼면 훨씬 간결해질 것 같네요 감사합니다